examples/quickstart/matrix_multiplication.cpp

```diff
@@ -0,0 +1,87 @@
+////////////////////////////////////////////////////////////////////////////////
+//  Copyright (c) 2021 Dimitra Karatza
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+////////////////////////////////////////////////////////////////////////////////
+
+// Parallel implementation of matrix multiplication
+
+#include <hpx/local/algorithm.hpp>
+#include <hpx/local/future.hpp>
+#include <hpx/local/init.hpp>
+
+#include <iostream>
+#include <random>
+#include <vector>
+
+///////////////////////////////////////////////////////////////////////////////
+//[mul_hpx_main
+int hpx_main()
+{   
+    int rowsA, colsA, rowsB, colsB;
+    rowsA = 2;
+    colsA = 3;
+    rowsB = colsA;
+    colsB = 2;
+    
+    // Initialize matrices A and B
+    std::vector<int> A(rowsA * colsA);
+    std::vector<int> B(rowsB * colsB);
+    std::vector<int> R(rowsA * colsB);
+
+    // Matrices have random values in the range [0, 10]
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    std::uniform_real_distribution<double> dis(0.0, 10.0);
```

Comment:
By the way, this should use the same type as the matrix. It might be nice to put a `using element_type = int` at the top of the function and use that alias in place of the concrete type.

```suggestion
    std::uniform_real_distribution<element_type> dis(0, 10);
```